pkg/scheduler/algorithm/predicates/metadata.go

```diff
@@ -131,6 +147,27 @@ func (meta *predicateMetadata) RemovePod(deletedPod *v1.Pod) error {
 	}
 	// Delete any anti-affinity rule from the deletedPod.
 	delete(meta.matchingAntiAffinityTerms, deletedPodFullName)
+	// Delete pod from the matching affinity or anti-affinity pods if exists.
+	affinity := meta.pod.Spec.Affinity
+	podNodeName := deletedPod.Spec.NodeName
+	if affinity != nil && len(podNodeName) > 0 {
+		if affinity.PodAffinity != nil {
+			for i, p := range meta.matchingAffinityPods[podNodeName] {
+				if p == deletedPod {
```

Comment:
Shouldn't we use a `reflect.DeepEqual`?

Reply:
It is not needed as the removal can happen only within a single scheduling cycle. We take a snapshot of the NodeInfo at the start of a scheduling cycle and the pointers are expected to remain the same during the cycle for the take snapshot. But if we decide to change that logic one day, we should then compare UID of the pods.